### PR TITLE
1586 fix vt sample project segfault

### LINKED ISSUE #1586: Segfault when running vt-trace-only-sample from vt-sample-project
When compiled with `build_with_tpl=1`:
```bash
root@a942e6b00517:/build/vt/vt-sample-project/build# ./vt-trace-only-sample 
Segmentation fault (core dumped)
```

and with `build_with_libs=1`:
```bash
root@a942e6b00517:/# ./build/vt/vt-sample-project/build/vt-trace-only-sample 
AddressSanitizer:DEADLYSIGNAL
=================================================================
==5395==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000138 (pc 0x0000004d6efb bp 0x7ffe44286fe0 sp 0x7ffe44286d80 T0)
==5395==The signal is caused by a READ memory access.
==5395==Hint: address points to the zero page.
    #0 0x4d6efb in vt::trace::TraceLite::TraceLite(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) /vt/src/vt/trace/trace_lite.cc:128:42
    #1 0x4d5efa in main /build/vt/vt-sample-project/build/../vt-trace-only/src/sample.cc:24:26
    #2 0x7f034a4f90b2 in __libc_start_main /build/glibc-eX1tMB/glibc-2.31/csu/../csu/libc-start.c:308:16
    #3 0x42b05d in _start (/build/vt/vt-sample-project/build/vt-trace-only-sample+0x42b05d)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV /vt/src/vt/trace/trace_lite.cc:128:42 in vt::trace::TraceLite::TraceLite(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
==5395==ABORTING
```

### PR DESCRIPTION
**Pull request by jstrzebonski**
Fixes #1586 

----

`TraceLite` ctor tried to read from config pointer, which wasn't initialized yet.

### CODE REVIEW

File: src/vt/trace/trace_lite.cc [hunk 1]
```diff
@@ -125,25 +125,31 @@ TraceLite::TraceLite(std::string const& in_prog_name)
    *  incremental_flush_mode_ to Z_FINISH
    */
 
-  auto const use_z_finish = theConfig()->vt_trace_gzip_finish_flush;
-  setFlushType(use_z_finish ? Z_FINISH : Z_SYNC_FLUSH);
+  auto const* conf_ptr = theConfig();
+  if (conf_ptr) {
+    auto const use_z_finish = conf_ptr->vt_trace_gzip_finish_flush;
+    setFlushType(use_z_finish ? Z_FINISH : Z_SYNC_FLUSH);
+  }
```
Comment (jstrzebonski) [inline src/vt/trace/trace_lite.cc:132]: I wonder if that code should stay at all?
Is there a case for which `theConfig()` will return something else than nullptr at this point?

Comment (cz4rs) [reply src/vt/trace/trace_lite.cc:132]: Possibly related: https://github.com/DARMA-tasking/vt/blob/b92912381377d37ef9e37e5a7f511204b37da85c/src/vt/runtime/runtime.cc#L110-L136

File: ci/build_cpp.sh [hunk 2]
```diff
@@ -259,4 +259,8 @@ then
       "$inclusion_type" \
       ..
     cmake --build .
+
+    # Try to actually run samples
```
Comment (cz4rs) [inline ci/build_cpp.sh:263]: Do you want this to stay or that's just a debugging leftover?

File: src/vt/trace/trace_lite.cc (hunk 1 above)
Comment (JacobDomagala) [reply src/vt/trace/trace_lite.cc:132]: Yes, when vt is used in "normal" mode (not trace-only), then `theConfig()` should be valid.

File: ci/build_cpp.sh (hunk 2 above)
Comment (jstrzebonski) [reply ci/build_cpp.sh:263]: I want it to stay.
I think we'd like to know whether changes in don't break anything - and running vt sample seems like a good first, naive test.

File: src/vt/trace/trace_lite.cc (hunk 1 above)
Comment (jstrzebonski) [reply src/vt/trace/trace_lite.cc:132]: Sure, but `TraceLite` is only used for `trace_only` isn't it?

Comment (JacobDomagala) [reply src/vt/trace/trace_lite.cc:132]: vt::Trace inherits from TraceLite

Comment (jstrzebonski) [reply src/vt/trace/trace_lite.cc:132]: Oh, okay, I've overlooked that!

**Review by PhilMiller — APPROVED** [review]
Generally looks good to me, though Cezary maybe has more to say about it before it gets merged.

**Review by cz4rs — APPROVED** [review]
Looks good to me.

File: ci/build_cpp.sh (hunk 2 above)
Comment (cz4rs) [reply ci/build_cpp.sh:263]: I was wondering whether we should move that towards testing step, but in the end I think these are fine here.

**Pull request merged**